### PR TITLE
Add support for `HIGHEST` priority to transport Events.

### PR DESCRIPTION
**Pull request by vkryachko**

### CODE REVIEW

File: transport/transport-api/src/main/java/com/google/android/datatransport/Priority.java [hunk 1]
```diff
@@ -14,10 +14,41 @@
...
   /** Quality of the service is within an hour. */
-  DEFAULT,
+  DEFAULT(0),
+
+  /** Events delivered at most daily, on unmetered network. */
+  VERY_LOW(1),
+
+  /** Events delivered as soon as possible. */
+  HIGHEST(2),
+  ;
```
Comment (rlazo) [inline transport/transport-api/src/main/java/com/google/android/datatransport/Priority.java]: remove the `;`

File: transport/transport-api/src/main/java/com/google/android/datatransport/Priority.java [hunk 2]
```diff
@@ -14,10 +14,41 @@
...
+  }
+
+  public int getValue() {
+    return value;
+  }
 
-  /** Events delivered at most daily, on unmetered network */
-  VERY_LOW,
+  public static Priority forValue(int value) {
+    if (value < 0 || value >= PRIORITY_MAP.length) {
```
Comment (rlazo) [inline transport/transport-api/src/main/java/com/google/android/datatransport/Priority.java]: For this to work, values would have to increment monotonically.  Either comment that in the `enum`, or use check the last value of the array as max.

File: transport/transport-api/src/main/java/com/google/android/datatransport/Priority.java (hunk 1 above)
Comment (vkryachko) [reply transport/transport-api/src/main/java/com/google/android/datatransport/Priority.java]: It's needed syntactically actually, note the comma one line above. I added it to make these lines friendlier to future diffs

Comment (rlazo) [reply transport/transport-api/src/main/java/com/google/android/datatransport/Priority.java]: ack

File: transport/transport-api/src/main/java/com/google/android/datatransport/Priority.java (hunk 2 above)
Comment (vkryachko) [reply transport/transport-api/src/main/java/com/google/android/datatransport/Priority.java]: I fixed the underlying issue that you pointed out. Note that I also removed the int -> priority logic into transport-runtime.

**Pull request merged**